metaflow/plugins/kfp/kfp_utils.py

```diff
@@ -0,0 +1,171 @@
+    return trigger_flow_by_id(
+        pipeline_id=pipeline_id,
+        experiment_name=experiment_name,
+        namespace=namespace,
+        triggerred_flow_name=triggerred_flow_name,
+        pipeline_parameters=pipeline_parameters,
+        pipeline_version=pipeline_version,
+    )
+
+
+def trigger_flow_by_id(
+    pipeline_id: str,
+    triggerred_flow_name: str,
+    namespace: str,
+    experiment_name: str = None,
+    pipeline_parameters: dict = None,
+    pipeline_version: str = None,
+) -> str:
+    """Trigger KFP flow by pipeline id.
+    pipeline_id: Pipeline id for which a new run should be triggered.
+    experiment_name: Experiment where the triggered run should be placed.
+    job_name: Job name of the flow being triggered.
+
+    Return run id of created run. To reconstruct url see run_id_to_url function.
+    """
+    client = _get_kfp_client()
+
+    if pipeline_parameters is None:
+        pipeline_parameters = {}
+
+    # Not checking for experiment existence: kfp client does not recreate experiment if exists
+    experiment: kfp_server_api.ApiExperiment = client.create_experiment(
+        name=experiment_name,
+        description="Experiment flow trigger from same namespace",
+        namespace=namespace,  # TODO: Warn about permission issue early
+    )
+
+    pipeline_run: kfp_server_api.ApiRun = client.run_pipeline(
+        experiment_id=experiment.id,
+        job_name=triggerred_flow_name,
+        pipeline_id=pipeline_id,
+        params={"flow_parameters_json": json.dumps(pipeline_parameters)},
+        version_id=pipeline_version,
+    )
+
+    return pipeline_run.id
+
+
+def run_id_to_url(run_id: str):
+    return posixpath.join(
+        KFP_RUN_URL_PREFIX,
+        "_/pipeline/#/runs/details",
+        run_id,
+    )
+
+
+def check_kfp_run_status(
+    run_id: str, timeout: int = -1
+) -> (bool, bool, kfp_server_api.ApiRun):
+    """Check for KFP run status.
+
+    If timeout (in second) is positive integers this function waits for flow to complete first.
+    Return tuple containing
+    - finished or not (bool)
+    - success or not (bool)
+    - run info (kfp_server_api.ApiRun)
+
+    TODO before merge: Print v.s. Logging
+        - KFP client use logging.info which may be silenced by default
+        - metaflow log formatter for splunk digestion?
+    """
+    client: kfp.Client = _get_kfp_client()
+    if timeout > 0:
+        print(f"Waiting for run {run_id} to finish. Timeout: {timeout}")
+        run: kfp_server_api.ApiRun = client.wait_for_run_completion(
+            run_id, timeout=timeout
+        ).run
+    else:
+        run: kfp_server_api.ApiRun = client.get_run(run_id).run
+
+    finished = run.status.lower() in ["succeeded", "failed", "skipped", "error"]
+    succeeded = run.status.lower() == "succeeded"
+
+    print(
+        f"Run {run_id} {'finished' if finished else 'did not finish'} with status {run.status}"
+    )
+
+    return succeeded, finished, run
```

Comment:
what happens if it times out?

Reply:
Changed function to return only `run` object, and raise `TimeoutError` on time out.

Deliberated on return value v.s. exception:
Returning succeeded and finished are basically returning error code, forcing customers to check for values, which is not very convenient or straight forward.

Change committed 🚀